src/main/java/de/tum/in/www1/artemis/service/CompetencyProgressService.java

```diff
@@ -265,4 +269,16 @@ else if (learningObject instanceof Exercise exercise) {
         throw new IllegalArgumentException("Learning object must be either LectureUnit or Exercise");
     }
 
+    /**
+     * Checks if the user associated to this {@code CompetencyProgress} has mastered the associated {@code Competency}.
+     *
+     * @param competencyProgress The user's progress
+     * @return True if the user mastered the competency, false otherwise
+     */
+    public static boolean isMastered(@NotNull CompetencyProgress competencyProgress) {
+        final double weight = 2.0 / 3.0;
```

Comment:
Where does this number come from?

Reply:
We use this weight in the client to determine the mastery.